### PR TITLE
Remove old Data analysis tab classes

### PR DESCRIPTION
**Pull request by robertapplin**
### Description of work
This PR removes the `ConvFitTab`, `MSDFitTab`, `IqtFitTab` and `FqFitTab` classes. Since the move to MVP a few years ago in the Data Analysis interface, all the functions for the tabs should be in the relevant Model-View-Presenter files for the relevant section of the GUI. This PR finally removes these classes so that new functionality is not added into them anymore.

Some of the changes might look a bit messy, but this is because this is only a stepping stone. There is still a lot of refactoring to do in this area, in particular surrounding the `IndirectDataAnalysisTab` class. I will continue with this refactor in a separate PR so that this one does not get too large. I have opened an issue to continue refactoring #36390

*There is no associated issue.*

### To test:
1. Go to `Interfaces`
2. Then `Inelastic Data Analysis`
4. Select `Add Workspace` for each tab and enter the data below for the corresponding tabs
5. Select the corresponding fit functions
6. Click `Run`
7. Click `Plot Spectra` or `Plot` (depending on the tab). Try `Plot Tiled` on the `Iqt` tab (accessed via the down arrow on the `Plot Spectra` button)

Also try moving around the range selectors on the embeded plots (the dashed vertical lines).

### **Test Data**

**MSDFit Tab**
Input file: [osi92762_graphite002_elwin_eq.zip](https://github.com/mantidproject/mantid/files/2361050/osi92762_graphite002_elwin_eq.zip)
Select Fit Type: `MSDGauss`

**IqtFit Tab**
Input file: [irs26176_graphite002_iqt.zip](https://github.com/mantidproject/mantid/files/2361062/irs26176_graphite002_iqt.zip)
Tick `Stretched Exponential`
Select `Flat background`
Note: drag the EndX on the graph to the left a bit to miss the noise (`0.15` I think)

**ConvFit Tab**
Input files: [irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2361070/irs26176_graphite002_red.zip)
[irs26173_graphite002_res.zip](https://github.com/mantidproject/mantid/files/2361071/irs26173_graphite002_res.zip)
Select Fit Type: `Two Lorenztian`

**F(Q) Tab**
Input file: [IN16B_125878_QLd_Result.zip](https://github.com/mantidproject/mantid/files/2361073/IN16B_125878_QLd_Result.zip)
Select Fit Type: `Chudley Elliott`

*This does not require release notes* because **it is an internal refactor**

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: qt/scientific_interfaces/Inelastic/Analysis/IndirectFitOutputOptionsView.cpp [hunk 1]
```diff
@@ -20,6 +20,8 @@ IndirectFitOutputOptionsView::IndirectFitOutputOptionsView(QWidget *parent)
   connect(m_outputOptions->pbPlot, SIGNAL(clicked()), this, SLOT(emitPlotClicked()));
   connect(m_outputOptions->pbSave, SIGNAL(clicked()), this, SLOT(emitSaveClicked()));
   connect(m_outputOptions->pbEditResult, SIGNAL(clicked()), this, SLOT(emitEditResultClicked()));
+
+  m_outputOptions->pbEditResult->setVisible(false);
```
Comment (MohamedAlmaki) [inline qt/scientific_interfaces/Inelastic/Analysis/IndirectFitOutputOptionsView.cpp:24]: Why this change is introduced?

File: qt/scientific_interfaces/Inelastic/Analysis/IndirectDataAnalysisTab.h [hunk 2]
```diff
@@ -35,13 +37,36 @@ class MANTIDQT_INELASTIC_DLL IndirectDataAnalysisTab : public IndirectTab {
...
 public:
-  IndirectDataAnalysisTab(IndirectFittingModel *model, QWidget *parent = nullptr);
+  IndirectDataAnalysisTab(std::string const &tabName, bool const hasResolution, QWidget *parent = nullptr);
   virtual ~IndirectDataAnalysisTab() override = default;
 
-  void setFitDataPresenter(std::unique_ptr<IndirectFitDataPresenter> presenter);
-  void setPlotView(IIndirectFitPlotView *view);
-  void setOutputOptionsView(IIndirectFitOutputOptionsView *view);
-  void setFitPropertyBrowser(IndirectFitPropertyBrowser *browser);
+  template <typename FittingModel> void setupFittingModel() { m_fittingModel = std::make_unique<FittingModel>(); }
```
Comment (MohamedAlmaki) [inline qt/scientific_interfaces/Inelastic/Analysis/IndirectDataAnalysisTab.h:43]: I think it is better to move these functions to the cpp file to keep the header file minimal

**Review by MohamedAlmaki — COMMENTED** [review]
Thanks for the refactoring, now the code is cleaner. Just a few comments

File: qt/scientific_interfaces/Inelastic/Analysis/FunctionBrowser/FqFunctionModel.h [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include "Analysis/FunctionBrowser/SingleFunctionTemplateModel.h"
+
+namespace MantidQt {
+namespace CustomInterfaces {
+namespace IDA {
+
+class FqFunctionModel : public SingleFunctionTemplateModel {
```
[the remark this answers is not in the document]
Comment (robertapplin) [reply qt/scientific_interfaces/Inelastic/Analysis/FunctionBrowser/FqFunctionModel.h:15]: Yes this is intentional. From what I can see, the FqFit `getEstimationDataSelector` only differs from the new parent function in that the tolerance value has changed from 1e-7 to 1e-5 (an intentional change)

With regards to MSDFit, thanks for pointing this out. I think I have missed moving the `if (xmin > xmax)` check to the parent function - I will make this change

File: qt/scientific_interfaces/Inelastic/Analysis/IndirectFitOutputOptionsView.cpp (hunk 1 above)
Comment (robertapplin) [reply qt/scientific_interfaces/Inelastic/Analysis/IndirectFitOutputOptionsView.cpp]: The `Edit Result` option is hidden for the MSDFit and FQFit tabs. It looks like I have forgot to show it for the ConvFit and IqtFit tabs so thanks for spotting this. It should now be fixed in the latest commits

File: qt/scientific_interfaces/Inelastic/Analysis/IndirectDataAnalysisTab.h (hunk 2 above)
Comment (robertapplin) [reply qt/scientific_interfaces/Inelastic/Analysis/IndirectDataAnalysisTab.h:43]: I have moved the non-templated setup function to the cpp file. The templated setup member functions cannot be implemented in the cpp file because the compiler requires to see the entire definition of a templated function, and it can also see if when it is in the header file. I am hoping I can tidy this up, or find a solution, as part of #36390

Comment (MohamedAlmaki) [reply qt/scientific_interfaces/Inelastic/Analysis/IndirectDataAnalysisTab.h:43]: Ok, thanks for the explanation!

**Review by MohamedAlmaki — APPROVED** [review]
Tabs work as expected. Thanks for enhancing the code quality
